sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -704,22 +1156,138 @@ pi_result piMemRetain(pi_mem) {
 }
 
 pi_result piMemRelease(pi_mem Mem) {
-  (void)Mem;
-  DIE_NO_IMPLEMENTATION;
+  if (Mem->getMemType() == PI_MEM_TYPE_BUFFER) {
```

Comment:
implementing piMemRelease as unconditional destruction of underlying CM memory object is extremely unsafe and error prone. E.g. there can be two kernels running in parallel using a mem buffer, one releases the buffer after it is done, the second is still using it.

Suggest to implement `piMemRetain` and ref counter checking in piMemRelease. @smaslov-intel - am I missing something?

Reply:
`piMemRetain` is implemented - RefCounter is used.